app/src/androidTest/kotlin/e2e/pageObjects/Page.kt

```diff
@@ -0,0 +1,122 @@
+package e2e.pageObjects
+
+import androidx.compose.ui.test.SemanticsNodeInteraction
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.assertIsEnabled
+import androidx.compose.ui.test.assertIsNotEnabled
+import androidx.compose.ui.test.assertTextEquals
+import androidx.compose.ui.test.junit4.ComposeTestRule
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+
+/**
+ * Base class for all page objects in the Bitwarden app.
+ * Provides a shared ComposeTestRule instance for UI testing.
+ */
+abstract class Page(protected val composeTestRule: ComposeTestRule) {
+    companion object {
+       const val TIMEOUT_MILLIS = 30000L
+    }
+
+    /**
+     * Waits for an element with the specified test tag to be present.
+     * @param testTag The test tag of the element to wait for
+     * @return SemanticsNodeInteraction for the found element
+     * @throws AssertionError if the element is not found within the timeout period
+     */
+    protected fun getElement(testTag: String): SemanticsNodeInteraction {
+        waitForIdle()
+        waitUntil(TIMEOUT_MILLIS) {
+            try {
+                composeTestRule.onNodeWithTag(testTag).assertExists()
+                true
+            } catch (e: AssertionError) {
+                false
+            }
+        }
+        return composeTestRule.onNodeWithTag(testTag)
+    }
+
+    protected fun getElementByText(text: String): SemanticsNodeInteraction {
+        waitForIdle()
+        waitUntil(TIMEOUT_MILLIS) {
+            try {
+                composeTestRule.onNodeWithText(text).assertExists()
+                true
+            } catch (e: AssertionError) {
+                false
+            }
+        }
+        return composeTestRule.onNodeWithText(text)
+    }
+
+    /**
+     * Waits for the app to be idle before proceeding with any UI interactions.
+     * This helps prevent flaky tests by ensuring the UI is stable.
+     */
+    protected fun waitForIdle() {
+        composeTestRule.waitForIdle()
+    }
+
+    /**
+     * Waits for a specific condition to be true before proceeding.
+     * @param timeoutMillis Maximum time to wait in milliseconds
+     * @param condition The condition to wait for
+     */
+    protected fun waitUntil(
+        timeoutMillis: Long,
```

Comment:
⛏️ 🎨 We could declare `TIMEOUT_MILLIS` as the default to avoid passing it with every call.
```suggestion
        timeoutMillis: Long = TIMEOUT_MILLIS,
```